File: Intermediate_Work/Massa_Lab07/Die.java
/**
 * @Author Dani Massa
 *
 * I affirm that I have carried out the attached academic endeavors with full academic honesty,
 * in accordance with the Union College Honor Code and the course syllabus.
 *
 */

import java.util.Random;
public class Die {
    private int numberOfSides;
    private int currentValue;

    /**
     * constructor for a die object
     * @param newNumberOfSides indicates the number of sides the die should have.
     */
    public Die(int newNumberOfSides) {
        numberOfSides = newNumberOfSides;
        currentValue = newNumberOfSides;
    }

    public Die (){
        numberOfSides = 6;
        currentValue = 6;
    }

    /**
     * reassigns the current value property to be a new random number, from 1 to the number of sides.
     */
    public void roll(){
        Random randomInt = new Random ();
        this.currentValue = randomInt.nextInt(numberOfSides)+1;
    }

    /**
     * returns the value currently showing on the die
     * @return currentValue
     */
    public int getValue (){
        return currentValue;
    }

    public int getNumberOfSides (){
        return numberOfSides;
    }


    public String toString(){
        return this.getNumberOfSides() + " sided die showing " + this.getValue();
    }
}

File: Intermediate_Work/Massa_Lab04/die.py
"""
defines the properties and behavior of a single playing die.
:author Dani Massa
"""
import random

CURRENT_VALUE = 6


# this constant is used to give the value variable within the constructor some arbitrary value. The value variable
# could have also been set equal to the parameter sides.


class Die:

    def __init__(self, sides=6):  # if no parameter is given, the default number of sides for the die is 6.
        """
        constructor
        :param sides: the number of sides the die has
        """
        self.__sides = sides
        self.__value = CURRENT_VALUE
        # value is the current value showing on the die

    def roll(self):
        """
        rolls a die by reassigning its current value property to a random number between 1 and its number of sides.
        """
        self.__value = random.randint(1, self.__sides)

    def get_value(self):
        """
        returns the value currently showing on the die.
        """
        return self.__value
